--- sysklogd-1.5/Makefile.orig	2007-05-30 17:28:48.000000000 +0200
+++ sysklogd-1.5/Makefile	2007-09-03 23:46:08.526200021 +0200
@@ -20,7 +20,7 @@
 CC= gcc
 #SKFLAGS= -g -DSYSV -Wall
 #LDFLAGS= -g
-SKFLAGS= $(RPM_OPT_FLAGS) -O3 -DSYSV -fomit-frame-pointer -Wall -fno-strength-reduce
+SKFLAGS= $(OPTIMIZE) -DSYSV -Wall $(shell getconf LFS_CFLAGS)
 # -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 # -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE
 # $(shell getconf LFS_SKFLAGS)
