diff -Nur sysklogd-1.4.1.orig/syslog_tst.c sysklogd-1.4.1/syslog_tst.c
--- sysklogd-1.4.1.orig/syslog_tst.c	Mon Jun  2 19:21:41 1997
+++ sysklogd-1.4.1/syslog_tst.c	Fri Mar 16 13:34:54 2001
@@ -42,7 +42,7 @@
 					if ( (nl = strrchr(bufr, '\n')) != \
 					    (char *) 0)
 						*nl = '\0';
-					syslog(LOG_INFO, bufr);
+					syslog(LOG_INFO, "%s", bufr);
 					logged += strlen(bufr);
 					if ( logged > 1024 )
 					{
@@ -54,7 +54,7 @@
 		}
 		else
 			while (argc-- > 1)
-				syslog(LOG_INFO, argv++[1]);
+				syslog(LOG_INFO, "%s", argv++[1]);
 	}
 	else
 	{
