--- sysklogd-1.4.1/CHANGES	2001-03-11 21:35:51.000000000 +0200
+++ sysklogd/CHANGES	2005-01-15 21:14:21.000000000 +0200
@@ -1,3 +1,42 @@
+Version 1.4.2
+
+ . Dmitry V. Levin <ldv@altlinux.org>
+   - Close file descriptor in FindSymbolFile() in ksym.c in order not to
+     leak file descriptors.
+ . Solar Designer <solar@openwall.com>
+   - improve crunch_list()
+   - Prevent potential buffer overflow in reading messages from the
+     kernel log ringbuffer.
+   - Ensure that "len" is not placed in a register, and that the
+     endtty() signal handler is not installed too early which could
+     cause a segmentation fault or worse.
+ . Steve Grubb <linux_4ever@yahoo.com>
+   - fix memory calculation in crunch_list()
+ . Martin Schulze <joey@infodrom.org>
+   - klogd will reconnect to the logger (mostly syslogd) after it went
+     away
+   - On heavily loaded system syslog will not spit out error messages
+     anymore when recvfrom() results in EAGAIN
+   - Makefile improvements
+   - Local copy of module.h
+   - Improved sysklogd.8
+   - Always log with syslogd's timezone and locale
+   - Remove trailing newline when forwarding messages
+ . Jon Burgess <Jon_Burgess@eur.3com.com>
+   - Moved the installation of the signal handler up a little bit so it
+     guaranteed to be available when the child is forked, hence, fixing a
+     race condition.  This used to create problems with UML and fast
+     machines.
+ . Greg Trounson <gregt@maths.otago.ac.nz>
+   - Improved README.linux
+ . Ulf Härnhammar <Ulf.Harnhammar.9485@student.uu.se>
+   - Bondary check for fscanf() in InitKsyms() and CheckMapVersion()
+ . Colin Phipps <cph@cph.demon.co.uk>
+   - Don't block on the network socket in case of package los
+ . Dirk Mueller <mueller@kde.org>
+   - Don't crash when filesize limit is reached (e.g. without LFS)
+
+
 Version 1.4.1
 
  . klogd will set the console log level only if `-c' is given on the
@@ -30,3 +69,9 @@
  . Olaf Kirch <okir@caldera.de>
    - Remove Unix Domain Sockets and switch to Datagram Unix Sockets
  . Several bugfixes and improvements, please refer to the .c files
+
+
+Local variables:
+mode: indented-text
+fill-column: 72
+End:
--- sysklogd-1.4.1/Makefile	1998-10-12 23:25:15.000000000 +0300
+++ sysklogd/Makefile	2006-05-25 10:12:44.000000000 +0300
@@ -4,12 +4,17 @@
 #CFLAGS= -g -DSYSV -Wall
 #LDFLAGS= -g
 CFLAGS= $(RPM_OPT_FLAGS) -O3 -DSYSV -fomit-frame-pointer -Wall -fno-strength-reduce
+# -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+# -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE
+# $(shell getconf LFS_CFLAGS)
 LDFLAGS= -s
 
 # Look where your install program is.
 INSTALL = /usr/bin/install
-BINDIR = /usr/sbin
-MANDIR = /usr/man
+
+# Destination paths, set prefix=/opt if required
+BINDIR = $(prefix)/usr/sbin
+MANDIR = $(prefix)/usr/share/man
 
 # There is one report that under an all ELF system there may be a need to
 # explicilty link with libresolv.a.  If linking syslogd fails you may wish
@@ -34,8 +39,9 @@
 # The following define establishes ownership for the man pages.
 # Avery tells me that there is a difference between Debian and
 # Slackware.  Rather than choose sides I am leaving it up to the user.
-MAN_OWNER = root
-# MAN_OWNER = man
+MAN_USER = root
+MAN_GROUP = root
+MAN_PERMS = 644
 
 # The following define establishes the name of the pid file for the
 # syslogd daemon.  The library include file (paths.h) defines the
@@ -116,7 +122,7 @@
 	${INSTALL} -m 500 -s klogd ${BINDIR}/klogd
 
 install_man:
-	${INSTALL} -o ${MAN_OWNER} -g ${MAN_OWNER} -m 644 sysklogd.8 ${MANDIR}/man8/sysklogd.8
-	${INSTALL} -o ${MAN_OWNER} -g ${MAN_OWNER} -m 644 syslogd.8 ${MANDIR}/man8/syslogd.8
-	${INSTALL} -o ${MAN_OWNER} -g ${MAN_OWNER} -m 644 syslog.conf.5 ${MANDIR}/man5/syslog.conf.5
-	${INSTALL} -o ${MAN_OWNER} -g ${MAN_OWNER} -m 644 klogd.8 ${MANDIR}/man8/klogd.8
+	${INSTALL} -o ${MAN_USER} -g ${MAN_GROUP} -m ${MAN_PERMS} sysklogd.8 ${MANDIR}/man8/sysklogd.8
+	${INSTALL} -o ${MAN_USER} -g ${MAN_GROUP} -m ${MAN_PERMS} syslogd.8 ${MANDIR}/man8/syslogd.8
+	${INSTALL} -o ${MAN_USER} -g ${MAN_GROUP} -m ${MAN_PERMS} syslog.conf.5 ${MANDIR}/man5/syslog.conf.5
+	${INSTALL} -o ${MAN_USER} -g ${MAN_GROUP} -m ${MAN_PERMS} klogd.8 ${MANDIR}/man8/klogd.8
--- sysklogd-1.4.1/README.1st	1997-06-02 20:21:39.000000000 +0300
+++ sysklogd/README.1st	2006-05-25 11:11:49.000000000 +0300
@@ -1,5 +1,5 @@
-Very important information before using version 1.3
----------------------------------------------------
+Important information
+---------------------
 
 The included version of syslogd behaves in a slightly different manner
 to the one in former releases.  Please review the following important
@@ -63,3 +63,33 @@
   these scripts should remove all old .pid files found in /var/run.
   This will insure that klogd and syslogd start properly even if prior
   executions have been terminated harshly.
+
+* Large file support, i.e. support to write to log files that are
+  larger than 2 GB is not part of syslogd, but a matter of the glibc
+  emitting different system calls to the kernel interface.  To support
+  large files you'll have to compile syslogd with the compiler defines
+  commented out in the main Makefile so that glibc adjusts the system
+  calls.
+
+Modifying the kernel console log level
+--------------------------------------
+
+  There are two ways to alter the kernel console log level.  This
+  setting controls whether log messages from the kernel should appear
+  on the system console or not.
+
+  In the past, klogd had to do this with the -c parameter.  Using '-c
+  4' will set the log level of console messages to 4 and only display
+  warnings and errors but not regular debug or information messages.
+
+  This behaviour is deprecated and hencely not enforced anymore via
+  the RC script of klogd.  Instead sysctl(8) should be used as
+  interface to various kernel variables.  These can be stored
+  non-volatile in /etc/sysctl.conf.
+
+  The prevent the kernel to flood the system console and to achieve
+  the same behaviour of '-c 4' simply add the following to the
+  configuration file and let sysctl set this kernel parameter upon
+  system boot.
+
+  kernel/printk = 4 4 1 7
--- sysklogd-1.4.1/README.linux	1999-01-19 02:09:12.000000000 +0200
+++ sysklogd/README.linux	2004-07-09 20:22:29.000000000 +0300
@@ -40,12 +40,17 @@
 a useful addition to the software gene pool.
 
 There is a mailing list covering this package and syslog in general.
-The lists address is sysklogd@Infodrom.North.DE .  To subscribe send a
-mail to Majordomo@Infodrom.North.DE with a line "subscribe sysklogd"
+The lists address is infodrom-sysklogd@lists.infodrom.org .  To subscribe send a
+mail to majordomo@lists.infodrom.org with a line "subscribe infodrom-sysklogd"
 in the message body.
 
-New versions of this package will be available at Joey's ftp server.
-ftp://ftp.infodrom.north.de/pub/people/joey/sysklogd/
+A second mailing list exists as infodrom-sysklogd-cvs@lists.infodrom.org.  Only
+CVS messages and diffs are distributed there. Whenever new code is added to
+sysklogd, CVS generates a mail from these changes which will be sent to
+this list.  Discussions will take place on the first list.
+
+The latest version of this software can be found at:
+http://www.infodrom.org/projects/sysklogd/download.php3
 
 Best regards,
 
@@ -67,6 +72,6 @@
 
 Martin Schulze
 Infodrom Oldenburg
-joey@linux.de
+joey@infodrom.org
 
-And a host of bug reporters whose contributions cannot be underestimated.
+And a number of bug reporters whose contributions cannot be underestimated.
--- sysklogd-1.4.1/debian/changelog	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/changelog	2001-04-21 15:06:18.000000000 +0200
@@ -0,0 +1,463 @@
+sysklogd (1.4.1-2) unstable; urgency=low
+
+  * Corrected location of GPL (closes: Bug#90582)
+  * Added section and priority for binary packages
+  * Added dependency to klogd so people who upgrade their sysklogd package
+    won't lose it anymore (closes: Bug#93922)
+  * Added code snipped to stop klogd/syslogd upon removal (closes:
+    Bug#90534, Bug#90970)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sat, 21 Apr 2001 15:06:02 +0200
+
+sysklogd (1.4.1-1) unstable; urgency=low
+
+  * New upstream source (closes: Bug#87819, Bug#70790, Bug#32580,
+    Bug#62358, Bug#71631)
+  * Upstream: Doesn't re-set log-level if not requested (closes:
+    Bug#76170, Bug#76170, Bug#85289)
+  * Upstream: Ignore zero bytes (closes: Bug#85478, Bug#85478, Bug#41068)
+  * Upstream: Corrected documentation for `-s' (closes: Bug#87020)
+  * Upstream: test for existence of syslogd-listfiles before calling
+    them.  This got lost due to 1.4.0 brokennes which was packaged and
+    removed some hours later (closes: Bug#84872, Bug#66712)
+  * Applied patch by Tommi Virtanen <tv@debian.org> splitting the package
+    into `sysklogd' and `klogd' (closes:Bug#35586, Bug#72043, Bug#74864,
+    Bug#72122)
+  * Provide / depend on virtual packages system-log-daemon
+    and linux-kernel-log-daemon (closes: Bug#67604)
+  * Applied patch from Tim Janik <timj@gtk.org> to support `-s pattern' in
+    syslogd-listfiles
+  * Transition to FHS, i.e. /usr/share/doc instead of /usr/doc and
+    /usr/share/man instead of /usr/man (closes:  Bug#79250, Bug#80771)
+  * Use --exec for stopping services (closes: Bug#76757)
+  * Corrected broken character in klogd.8 (cloes: Bug#75932)
+  * Only rotate logfiles with size greater than zero.  This got lost due
+    to 1.4.0 brokennes which was packaged and removed some hours later
+    (closes: Bug#74993, Bug#49824)
+  * Added another note about modificability of cronjobs (closes:
+    Bug#88741)
+  * Since klogd replaces parts of sysklogd a proper Replaces line is there
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 11 Mar 2001 22:30:53 +0100
+
+sysklogd (1.4-1) stable unstable; urgency=medium, closes=70790 32580 49824 62358 66712 71631
+
+  * Added paragraph to binary target
+  * New upstream version (closes: Bug#70790, Bug#32580, Bug#62358,
+    Bug#71631)
+  * Only rotate logfiles that have a size greater than zero (closes:
+    Bug#49824)
+  * Test for existence of syslogd-listfiles before calling it (closes:
+    Bug#66712)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Mon, 18 Sep 2000 16:02:05 +0200
+
+sysklogd (1.3-33) unstable; urgency=medium, closes=38977
+
+  * chmod 0640 /dev/xconsole regardless of its existence (closes: Bug#38977)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 12 Sep 1999 20:47:53 +0200
+
+sysklogd (1.3-32) unstable; urgency=low, closes=35408 34733 36198
+
+  * Fixed typo in syslogd-listfiles (closes: Bug#35408)
+  * Removed superflous call to utmpname() (closes: Bug#34733)
+  * Added --ignore-size and limit for rotating log files at 2MB to
+    syslogd-listfiles (closes: Bug#36198)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 23 May 1999 10:40:33 +0200
+
+sysklogd (1.3-31) frozen unstable; urgency=low, closes=28629 30093 31494 30462
+
+  * ``-m 0'' turns off -- MARK -- now. (closes: Bug#28629, Bug#31494)
+  * Changed Greg's e-mail address to represent his current one.
+  * Shortened line length for kernel logging slightly.
+  * Corrected return value of AddModule (closes: Bug#30093)
+  * Finally fixed an error with `-a' processing, thanks to Topi Miettinen
+    <tom@medialab.sonera.net> (closes: Bug#30462)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Tue, 19 Jan 1999 23:56:07 +0100
+
+sysklogd (1.3-30) unstable; urgency=low, closes=28128 28122 24893
+
+  * Contains more patches from Topi Miettinen.
+  * Fixed two mistakes from the most recent version
+  * Fixed fd leak problem.  Thanks to Topi for detecting it
+  * Ouch!  Looks like the released package again is broken.  It worked
+    locally.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 18 Oct 1998 20:38:22 +0200
+
+sysklogd (1.3-29) unstable; urgency=low, closes=24893
+
+  * Re-Applied patch provided vom Topi Miettinen with regard to the people
+    from OpenBSD.  This provides the additional '-a' argument used for
+    specifying additional UNIX domain sockets to listen to.  This is been
+    used with chroot()'ed named's for example.  An example is described at
+    http://www.psionic.com/papers/dns.html.  This time the patch doesn't
+    stall syslogd.  Thanks to Topi Miettinen <tom@medialab.sonera.net>
+    (closes: Bug#24893)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sat, 17 Oct 1998 20:23:00 +0200
+
+sysklogd (1.3-28) unstable; urgency=low
+
+  * Added manpage for syslog-facility, also written by Raphaël Hertzog 
+  * Reverted support for multiple Unix domain sockets since it caused
+    syslogd to not accept more sockets after a certain (undefinitve)
+    time.  (opens: Bug#24893)
+  * The complete code is still present, search for `Bug#24893'
+  * Commented out `-a' in the syslogd.8 manpage
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 15 Oct 1998 19:35:26 +0200
+
+sysklogd (1.3-27) frozen unstable; urgency=low, closes=21212 11917 21606 24609 25835 26015 24648 20135 23502 24893 24894
+
+  * Postinst will create files with correct file modes (closes: Bug#21212)
+  * Made /dev/xconsole mode 0640 and root.adm (closes: Bug#11917)
+  * Changed priority of exit message from LOG_ERR to LOG_INFO (closes
+    Bug#21606)
+  * The code now is compilable without SYSLOG_INET again.  Thanks to
+    Enrik Berkhan <enrik@rzstud1.akk.org>.
+  * Added support for TESTING define which will turn syslogd into
+    stdio-mode used for debugging.
+  * Reworked the initialization/fork code.  Now the parent
+    process activates a signal handler which the daughter process will
+    raise if it is initialized.  Only after that one the parent process
+    may exit.  Otherwise klogd might try to flush its log cache while
+    syslogd can't receive the messages yet. (closes: Bug#24609)
+  * Fixed typo in /etc/syslog.conf (closes: Bug#25835)
+  * Modified pre- and postinst scripts to set +e before executing external
+    scripts.  This is a workaround against a bug in bash (Bug#23857).
+    Bash exports the errexit (-e) flag if allexport (-a) has been set. The
+    bug occurs even if allexport is being _re_set.  (closes: Bug#26015)
+  * Modified ksym::CheckVersion()
+    . Use shift to decode the kernel version
+    . Compare integers of kernel version
+    . extract major.minor.patch from utsname.release via sscanf()
+    The reason lays in possible use of kernel flavours which modify
+    utsname.release but no the Version_ symbol.  (closes: Bug#20135)
+  * Corrected klogd.c in order to support the new address space for late
+    2.1 kernels.  Some support for unsigned long in contrary to int was
+    missing.  (closes: Bug#23502)
+  * Applied patch provided vom Topi Miettinen with regard to the people
+    from OpenBSD.  This provides the additional '-a' argument used for
+    specifying additional UNIX domain sockets to listen to.  This is been
+    used with chroot()'ed named's for example.  An example is described at
+    http://www.psionic.com/papers/dns.html.  Thanks to Topi Miettinen
+    <tom@medialab.sonera.net> (closes: Bug#24893)
+  * Added support for `ftp' log facility which was introduced by glibc
+    version 2.  Thanks to netgod and Flood from #Debian.
+  * Code cleanups with regard to bsd -> posix transition and stronger
+    security (buffer length checking).  Thanks to Topi Miettinen
+    <tom@medialab.sonera.net> (closes: Bug#24894)
+    . index() --> strchr()
+    . sprintf() --> snprintf()
+    . bcopy() --> memcpy()
+    . bzero() --> memset()
+    . UNAMESZ --> UT_NAMESIZE
+    . sys_errlist --> strerror()
+  * Added support for setutent()/getutent()/endutend() instead of binary
+    reading the UTMP file.  This is the the most portable way.  Thanks to
+    Topi Miettinen <tom@medialab.sonera.net>.
+  * Avoid logging of SIGCHLD when syslogd is in the process of exiting and
+    closing its files.  Again thanks to Topi.
+  * Added /usr/sbin/syslog-facility, written by Raphaël Hertzog
+    <hertzog@debian.org>.  It opens a way for other packages to add and
+    remove logfiles/facilities.
+  * Modified printline() to support 8bit characters - such as russion
+    letters.  Thanks to Vladas Lapinskas <lapinskas@mail.iae.lt>.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Mon, 12 Oct 1998 22:24:04 +0200
+
+sysklogd (1.3-26) frozen unstable; urgency=low, closes=20903 20765 20556 20507 20647 20220
+
+  * Fixed typos in syslogd-listfiles(1), thanks to Francesco Potorti`
+    <F.Potorti@cnuce.cnr.it> (closes: Bug#20903)
+  * Corrected email address for Shane Alderton (closes: Bug#20765)
+  * Fixed manpage to not reflect reverse implementated SIGALRM signal
+    handler (closes: Bug#20647)
+  * Modified behaviour with including <sys/module.h> (closes: Bug#20556)
+  * Added correct prototype for llseek() (closes: Bug#20507, Bug#20220)
+  * Added more log information if problems occurr while reading a system
+    map file.
+  * Modified System.map read function to try all possible map files until
+    a file with matching version is found.
+  * Added Debian release to klogd
+  * Switched to fgets() as gets() is not buffer overrun secure.
+  * Modified loop for detecting the correct system map.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 16 Apr 1998 19:28:05 +0200
+
+sysklogd (1.3-25) unstable; urgency=low, closes=19454 19145 1914
+
+  * Corrected permissions on packaging control files (closes: Bug#19414)
+  * Removed kill statements from rc file (closes: Bug#19454, Bug#19145)
+  * Applied patch from Pedro Manuel Rodrigues <pmanuel@cindy.fe.up.pt> to
+    let klogd work togther with >2GB address space.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 19 Mar 1998 23:56:09 +0100
+
+sysklogd (1.3-24) unstable; urgency=low, closes=18124
+
+  * Fixed tiny mistake which refused the MARK facility to work properly
+    (closes: Bug#18124)
+  * Corrected Topi's patch as it prevented forwarding during startup due
+    to an unknown LogPort.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Wed, 25 Feb 1998 13:19:27 +0100
+
+sysklogd (1.3-23) unstable; urgency=low
+
+  * Corrected FSF's address (lintian)
+  * Implemented force-reload (lintian)
+  * Standards-Version: 2.4.0.0
+  * Applied patch from Topi Miettinen <Topi.Miettinen@ml.tele.fi> to open
+    the UDP socket only if it is really needed.
+  * Updated sysklogd(8) manpage
+  * Added notes to mailing list
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Tue, 24 Feb 1998 00:39:10 +0100
+
+sysklogd (1.3-22) unstable; urgency=low, closes=17192 17385 17601
+
+  * Included a real fix to the %'s problem (fixes: Bug#17192)
+  * Another patch for Linux/alpha by Christopher C Chimelis
+    <chris@classnet.med.miami.edu> (fixes: Bug#17385)
+  * Removed killall-statement from postinst.  Thanks to James Troup for
+    reporting it.
+  * Added chdir("/") to both daemons (fixes: Bug#17601)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 29 Jan 1998 21:56:40 +0100
+
+sysklogd (1.3-21) unstable; urgency=low, closes=17159
+
+  * Fixed small typo in syslogd-listfiles (Bug#17159)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 15 Jan 1998 23:22:57 +0100
+
+sysklogd (1.3-20) unstable; urgency=low, closes=17000
+
+  * Corrected some code that caused klogd to dump core when receiving some
+    special messages from 2.1.78.  Thanks to Chu-yeon Park
+    <kokids@doit.ajou.ac.kr> for informing me.
+  * Fixed bug that caused klogd to die if there is no System.map
+    available.
+  * Added -x switch to omit EIP translation and System.map evaluation.
+    Thanks to Florian La Roche <florian@knorke.saar.de>
+  * Fixed small bugs in F_FORW_UNKN meachanism.  Thanks to Torsten Neumann
+    <torsten@londo.rhein-main.de> for pointing me to it.
+  * Fixed problem with klogd not being able to be built on a kernel newer
+    than 2.1.18.  Worked in a patch from Alessandro Suardi <asuardi@uninetcom.it>
+  * Fixed small typo in syslogd-listfiles
+  * Corrected debug output concerning remote receiption
+  * Removed race condition in syslogd-listfiles
+  * Modified logfile detection routine to take care of double listed
+    logfiles.  Thanks to Roman Hodek <roman@debian.org> for providing an
+    appropriate patch.  (Bug#17000)
+  
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Tue, 13 Jan 1998 01:59:56 +0100
+
+sysklogd (1.3-19) unstable; urgency=low, closes=16643 16769 16796 16828 16148 14776
+
+  * Used better regexp for detecting sensitive data.  Thanks a lot to
+    Susanne Schmidt <banshee@linux.de> for providing it.
+  * Fixed ownership of /usr/doc/sysklogd/copyright and readme (Bug#16643)
+  * Corrected Standards-Version to 2.3.0.1 (Bug#16769)
+  * Reworked one line of an older patch because it prevented syslogd from
+    binding the socket with the result that no messages were forwarded to
+    other hosts.
+  * Changed the behaviour of klogd when receiving a terminate signal. Now
+    the program terminates immediately instead of completing the receipt of
+    a kernel message.  (Bug#16796, Bug#16828, Bug#16148)
+  * Noticed a bug which was closed by 1.3-18 (Bug#14776)
+  * Changed Maintainer address to joey@debian.org
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Fri, 9 Jan 1998 00:58:36 +0100
+
+sysklogd (1.3-18) unstable; urgency=low, closes=11731 12009 13506 14329 14521 14610 15840 15336
+
+  * Added test in postinst if user and group exist.  Christoph Lameter run
+    into trouble upgrading a RedHat system.
+  * Fixed little mistake which prevented klogd from accepting a console
+    log level of 8 to get <7> alias KERN_DEBUG displayed.
+  * Linked against libc6 (Bug#11731)
+  * Added SHELL=/bin/bash to rules as Herbert Xu suggested
+  * Corrected syslog.conf(5) manpage
+  * syslogd resets the ignore priority flag now.  Thanks to Herbert
+    Thielen. (Bug#12009)
+  * Fixed bug that caused syslogd to write into wrong files under some
+    race conditions.  Thanks to Herbet Xu. (Bug#13506)
+  * /var/log/auth.log will only be touch if needed (Bug#14329)
+  * debian/conffiles is treated like a document (Bug#14521)
+  * Added script for generic logfile detection and rotation and included
+    that in cron scripts, included also its manpage (Bug#14610)
+  * Modified restart messages in /etc/init.d/sysklogd (Bug#15840)
+  * klogd will first try to load /boot/System.map-$ver, then
+    /boot/System.map, then /System.map-$ver and after that /System.map
+  * Modified ExpandKadds() because there were some problems accessing
+    memory r/w.  (Bug#15336)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Fri, 2 Jan 1998 03:31:09 +0100
+
+sysklogd (1.3-17) unstable; urgency=low
+
+  * Fixed stupid bug which caused klogd to eat up 90% cpu time.
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Wed, 25 Jun 1997 15:22:29 +0200
+
+sysklogd (1.3-16) unstable; urgency=low
+
+  * Included Miquel's patches which allows continuation line support
+    for syslogd.  Corrected syslog.conf(5) manpage.  Improved
+    syslog.conf file.  Thanks to Miquel van Smoorenburg
+    <miquels@cistron.nl> for dropping in these patches.
+  * Added /var/log/syslog as a daily rotated logfile which contains
+    everything except auth.*
+  * ksym.c: Added #1 and #2 to some error messages in order to being able
+    to divide them (ulmo@Q.Net)
+  * Improved postinst script (Thanks to Miquel van Smorrenborg for ideas)
+  * klogd now also reads /System.map-<version> and
+    /boot/System.map-<version> (Bug#7210)
+  * syslogd,klogd: made DEBRELEASE optional to support a non-Debian
+    release as well.
+  * Corrected freeing of logfiles.
+  * Removed delay of 10 seconds. (Bug#8104, Bug#8400)
+  * Modified permissions of logfiles from 644 to 640 and 640 to 600 for
+    security aspects.
+  * Corrected syslog.conf(5) manpage (Bug#9779)
+  * Found upstream patches for 1.3.1 to 1.3.3 which are now included.
+    These also include some patches for glibc and Alpha.
+  * Included patch to modules package to support better debugging
+  * Added patch from Leland Olds which fixes a buffer overrun and improved
+    symbol lookup. (Bug#4875)
+  * Some more glibc patches made by Michael Alan Dorman
+    <mdorman@debian.org>. (Bug#8362, Bug#8763)
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Sun, 15 Jun 1997 19:37:13 +0200
+
+sysklogd (1.3-15) unstable; urgency=low
+
+  * Added /var/log/mail.log as a logfile
+  * Corrected Description field (Bug#7624)
+  * Added read statement in postinst script (Bug#8243)
+
+ -- Martin Schulze <joey@infodrom.north.de>  Wed, 26 Mar 1997 13:32:24 +0100
+
+sysklogd (1.3-14) unstable; urgency=low
+
+  * Debugging syslogd now shows facility.priority
+  * Minor fixes
+  * Modified syslogd.c to not kill itself which confuses bash 2.0
+
+ -- Martin Schulze <joey@infodrom.north.de>  Sun, 23 Feb 1997 12:23:55 +0100
+
+sysklogd (1.3-13) stable; urgency=low
+
+  * Made /etc/init.d/sysklogd more verbose
+  * Minor fixes
+  * Uncompressed copyright, but compressed /usr/doc/sysklogd-1.3.16/readme
+
+ -- Martin Schulze <joey@debian.org>  Sat, 8 Feb 1997 14:12:29 +0100
+
+sysklogd (1.3-12) stable; urgency=HIGH
+  
+  * Converted to Standards-Version 2.1.1.2
+  * init.d/sysklogd: added "sleep 1"
+  * syslogd won't hassle anymore if it can't access logfils. Patched by
+    me. This adresses Bug#5865.
+  * /etc/init.d/sysklogd: Modified so that one can use commandline
+    arguments for both syslogd and klogd.
+  * debian.rules: Installed ChangeLog
+
+ -- Martin Schulze <joey@finlandia.infodrom.north.de>  Thu, 30 Jan 1997 22:48:59 +0100
+
+Mon Now 25 10:36:01 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* modified preinst to work if /etc/init.d/sysklogd doesn't exist
+	because of a failure
+
+Mon Sep 10 10:36:01 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* ksyms.c: klogd will first look at /System.map (Bug#4403, #4459)
+
+Fri Aug 16 21:36:01 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* debian.rules: Changed permissions for syslogd and klogd to 755
+	(Bug#4049)
+
+Mon Aug 12 10:40:05 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Corrected permissions of documentation.  Thanks to Dan
+        Quinlan. (Bug#4090)
+
+Mon Aug  5 22:47:40 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* syslogd.c: If you haven't set your domainname, syslogd will dump
+	core upon startup.  Fixed.  Thanks to Leeland Lucius
+	<llucius@millcomm.com> for reporting and patching it. (Bug#3526)
+
+	* /etc/init.d/sysklogd, /etc/cron.weekly/sysklogd: slight
+	improvements.  Thanks to Bernd Eckenfels <ecki@tapac.inka.de>.
+
+Fri Aug  2 11:03:30 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* modified weekly cron script, thanks to Todd Tyrone Fries <todd@miango.com>
+
+Fri Jun 14 10:42:26 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* ksyms.c: increased the size of vstring, because since Linux
+	major versions are decoded with n * 65536 and we now have 2.0, the
+	field was too short.
+
+Tue May 28 01:02:01 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* syslogd.c: Corrected behaviour of blocking pipes - i.e. the
+	whole system hung.  Michael Nonweiler <mrn20@hermes.cam.ac.uk> has
+	sent us a patch to correct this.  A new logfile type F_PIPE has
+	been introduced.
+
+Thu May 23 08:39:54 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* postinst: Corrected killall-statement, Thanks to Miquel van
+	Smoorenborg for the hint
+
+Mon May 20 00:14:35 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* postinst: added a killall to remove an existing syslogd.  It
+	could exist because of Bug#2836.
+
+Sun May 12 13:36:55 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* removed auth.* from /dev/xconsole
+
+	* klogd won't log old messages anymore (Bug#2337)
+	I have applied the patch from Chris Hanson.
+
+	* I have put another start-stop-daemon call in the startup script
+	so the configs are reloaded on startup. This should
+	help. (Bug#2552)
+
+	* Sysklogd takes care of this and createst /dev/xconsole if it
+	doesn't exist during installation. (Bug#2683)
+
+	* I have corrected permissions of /dev/xconsole. (Bug#2837)
+
+	* defined /etc/cron.weekly/sysklogd as a conffile (Bug#2950, Bug#2960)
+
+	* introduced two binary packages: syslogd and sysklogd which
+	exclude each other
+
+Sun Apr 21 17:56:03 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Corrected preinst (Bug#2722, Bug#2725, Bug#2726)
+
+	* Linked against libc 5.2.18 (Bug#2723, Bug#2725)
+
+	* Corrected control file to fit with newer dpkgs
+
+Wed Apr 17 21:40:56 1996  Martin Schulze  <joey@finlandia.infodrom.north.de>
+
+	* Added Debian packaging information
+
+
--- sysklogd-1.4.1/debian/conffiles	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/conffiles	1997-06-11 00:06:53.000000000 +0300
@@ -0,0 +1,4 @@
+/etc/syslog.conf
+/etc/init.d/sysklogd
+/etc/cron.daily/sysklogd
+/etc/cron.weekly/sysklogd
--- sysklogd-1.4.1/debian/conffiles.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/conffiles.klogd	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1 @@
+/etc/init.d/klogd
--- sysklogd-1.4.1/debian/control	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/control	2001-04-21 15:06:18.000000000 +0200
@@ -0,0 +1,37 @@
+Source: sysklogd
+Section: base
+Priority: required
+Maintainer: Martin Schulze <joey@debian.org>
+Standards-Version: 3.2.0
+
+Package: sysklogd
+Section: base
+Priority: required
+Architecture: any
+Depends: ${shlibs:Depends}
+Conflicts: syslogd
+Provides: syslogd, system-log-daemon
+Replaces: syslogd
+Depends: klogd | linux-kernel-log-daemon
+Section: base
+Priority: required
+Description: System logging daemons
+ This package implements the system log daemon, which is an enhanced
+ version of the standard Berkeley utility program. It is responsible
+ for providing logging of messages received from programs and facilities
+ on the local host as well as from remote hosts.
+
+Package: klogd
+Section: base
+Priority: required
+Architecture: any
+Depends: ${shlibs:Depends}, sysklogd | system-log-daemon
+Conflicts: sysklogd (<= 1.3-33)
+Provides: linux-kernel-log-daemon
+Replaces: sysklogd
+Description: Kernel logging daemon
+ The klogd daemon listens to kernel message sources and is responsible
+ for prioritizing and processing operating system messages.  The klogd
+ daemon can run as a client of syslogd or optionally as a standalone
+ program.  Klogd can now be used to decode EIP addresses if it can
+ determine a System.map file.
--- sysklogd-1.4.1/debian/copyright	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/copyright	2001-04-21 15:06:18.000000000 +0200
@@ -0,0 +1,34 @@
+This is the Debian GNU/Linux prepackaged version of Linux' system and
+kernel logging daemons.
+
+This package was put together by Martin Schulze <joey@debian.org>,
+from sources obtained from:
+ tsx-11.mit.edu:/pub/sources/sbin/sysklogd-1.3.tar.gz
+
+Recent versions are Debian versions that may be also be found at
+  ftp://ftp.infodrom.north.de/pub/people/joey/sysklogd/
+
+syslogd is under Berkeley copyright, klogd is under GPL.
+
+   Copyright (C) 1994-96 Greg Wettstein <greg@wind.enjellic.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 dated June, 1991.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program;  if not, write to the Free Software
+   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111, USA.
+
+On Debian GNU/Linux systems, the complete text of the GNU General
+Public License can be found in `/usr/share/common-licenses/GPL'.
+
+There is a mailing list covering this package and syslog in general.
+The lists address is sysklogd@Infodrom.North.DE .  To subscribe send a
+mail to Majordomo@Infodrom.North.DE with a line "subscribe sysklogd"
+in the message body.
--- sysklogd-1.4.1/debian/cron.daily	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/cron.daily	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,42 @@
+#! /bin/sh
+
+# sysklogd	Cron script to rotate system log files daily.
+#
+#		If you want to rotate other logfiles daily, edit
+#		this script.  An easy way is to add them manually
+#		or to add -a to syslogd-listfiles and add some grep
+#		stuff.
+#
+#		This is a configration file.  You are invited to edit
+#		it and maintain it on your own.  You'll have to do
+#		that if you don't like the default policy
+#		wrt. rotating logfiles (i.e. with large logfiles
+#		weekly and daily rotation may interfere).  If you edit
+#		this file and don't let dpkg upgrade it, you have full
+#		control over it.  Please read the manpage to
+#		syslogd-listfiles.
+#
+#		Written by Martin Schulze <joey@debian.org>.
+#		$Id$
+
+test -x /usr/sbin/syslogd-listfiles || exit 0
+
+cd /var/log
+for LOG in `syslogd-listfiles`
+do
+   if [ -s $LOG ]; then
+      savelog -g adm -m 640 -u root -c 7 $LOG >/dev/null
+   fi
+done
+
+for LOG in `syslogd-listfiles --auth`
+do
+   if [ -s $LOG ]; then
+      chown root.adm $LOG
+      chmod o-rwx $LOG
+   fi
+done
+
+# Restart syslogd
+#
+/etc/init.d/sysklogd reload
--- sysklogd-1.4.1/debian/cron.weekly	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/cron.weekly	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,44 @@
+#! /bin/sh
+
+# sysklogd	Cron script to rotate system log files weekly.
+#
+#		If you want to rotate logfiles daily, edit
+#		this script and /etc/cron.daily/sysklogd to get
+#		the logfiles in sync (they must not occur in both
+#		files).  Do not delete this file itself, since
+#		it will then be replaced after an upgrade creating
+#		unwanted effects.
+#
+#		This is a configration file.  You are invited to edit
+#		it and maintain it on your own.  You'll have to do
+#		that if you don't like the default policy
+#		wrt. rotating logfiles (i.e. with large logfiles
+#		weekly and daily rotation may interfere).  If you edit
+#		this file and don't let dpkg upgrade it, you have full
+#		control over it.  Please read the manpage to
+#		syslogd-listfiles.
+#
+#		Written by Ian A. Murdock <imurdock@debian.org>.
+#		$Id$
+
+test -x /usr/sbin/syslogd-listfiles || exit 0
+
+cd /var/log
+for LOG in `syslogd-listfiles --weekly`
+do
+   if [ -s $LOG ]; then
+      savelog -g adm -m 640 -u root -c 4 $LOG >/dev/null
+   fi
+done
+
+for LOG in `syslogd-listfiles --auth`
+do
+   if [ -s $LOG ]; then
+      chown root.adm $LOG
+      chmod o-rwx $LOG
+   fi
+done
+
+# Restart syslogd
+#
+/etc/init.d/sysklogd reload
--- sysklogd-1.4.1/debian/postinst	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/postinst	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,85 @@
+#! /bin/sh
+
+set -e
+
+if [ -z "`ls /dev/xconsole 2> /dev/null`" ]; then
+    mknod -m 640 /dev/xconsole p
+else
+    chmod 0640 /dev/xconsole
+fi
+chown root.adm /dev/xconsole
+
+if [ ! -d /var/log/news ] \
+	&&  grep -q ^news: /etc/passwd \
+	&& grep -q ^news: /etc/group 
+then
+    mkdir /var/log/news
+    chmod 2755 /var/log/news
+    chown news.news /var/log/news
+fi
+
+if [ "$1" = "configure" ]
+then
+    case $2 in
+	1.2-*)
+cat<<EOT
+The default behaviour of syslogd has altered from 1.2 to 1.3.  By default
+*no* messages from the UDP port are accepted.
+
+Read the documentation in /usr/doc/sysklogd carefully.  Some important
+things have been changed!
+
+Press [Enter] to continue
+EOT
+	read xyz
+	;;
+  esac
+fi
+
+if [ "$1" = "configure" ]
+then
+    if [ -f /etc/init.d/sysklogd ]
+    then
+	set +e
+	sh /etc/init.d/sysklogd stop
+	set -e
+    fi
+
+    update-rc.d sysklogd defaults 10 90 >/dev/null
+
+    # Create logfiles with correct file modes
+    #
+    for LOG in `syslogd-listfiles` `syslogd-listfiles --auth`
+    do
+	if [ ! -f $LOG ]; then
+	    touch $LOG
+	fi
+	chown root.adm $LOG
+	chmod 640 $LOG
+    done
+
+    # Tidying up some old stuff
+    #
+    rm -f /etc/rc?.d/[SK]*.syslogd
+
+    # restarting daemon
+    #
+    if [ -f /etc/init.d/sysklogd ]
+    then
+	set +e
+	sh /etc/init.d/sysklogd start
+	set -e
+    fi
+
+    # Maintain transition to FHS
+    if [ "$1" = "configure" ]
+    then
+	if [ -d /usr/doc -a ! -e /usr/doc/sysklogd -a -d /usr/share/doc/sysklogd ]
+	then
+	    ln -sf ../share/doc/sysklogd /usr/doc/sysklogd
+	fi
+    fi
+fi
+
+
+exit 0
--- sysklogd-1.4.1/debian/postinst.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/postinst.klogd	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,35 @@
+#! /bin/sh
+
+set -e
+
+if [ "$1" = "configure" ]
+then
+    if [ -f /etc/init.d/klogd ]
+    then
+	set +e
+	sh /etc/init.d/klogd stop
+	set -e
+    fi
+
+    update-rc.d klogd defaults 11 89 >/dev/null
+
+    # restarting daemon
+    #
+    if [ -f /etc/init.d/klogd ]
+    then
+	set +e
+	sh /etc/init.d/klogd start
+	set -e
+    fi
+
+    # Maintain transition to FHS
+    if [ "$1" = "configure" ]
+    then
+	if [ -d /usr/doc -a ! -e /usr/doc/klogd -a -d /usr/share/doc/klogd ]
+	then
+	    ln -sf ../share/doc/klogd /usr/doc/klogd
+	fi
+    fi
+fi
+
+exit 0
--- sysklogd-1.4.1/debian/postrm	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/postrm	2001-04-21 15:06:18.000000000 +0200
@@ -0,0 +1,13 @@
+#! /bin/sh
+
+set -e
+
+if [ "$1" = "purge"  -o "$1" = "remove" ]
+then
+    sh /etc/init.d/sysklogd stop
+fi
+
+if [ "$1" = "purge" ]
+then
+    update-rc.d sysklogd remove >/dev/null
+fi
--- sysklogd-1.4.1/debian/postrm.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/postrm.klogd	2001-04-21 15:06:18.000000000 +0200
@@ -0,0 +1,13 @@
+#! /bin/sh
+
+set -e
+
+if [ "$1" = "purge"  -o "$1" = "remove" ]
+then
+    sh /etc/init.d/klogd stop
+fi
+
+if [ "$1" = "purge" ]
+then
+  update-rc.d klogd remove >/dev/null
+fi
--- sysklogd-1.4.1/debian/preinst	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/preinst	1998-10-12 19:59:20.000000000 +0300
@@ -0,0 +1,16 @@
+#! /bin/sh
+
+set -e
+
+dpkg --assert-support-predepends
+
+if [ "$1" = "upgrade" -a -f /etc/init.d/sysklogd ]; then
+    if [ -f /etc/init.d/sysklogd ]
+    then
+	set +e
+	sh /etc/init.d/sysklogd stop
+	set -e
+    fi
+fi
+
+exit 0
--- sysklogd-1.4.1/debian/preinst.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/preinst.klogd	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,16 @@
+#! /bin/sh
+
+set -e
+
+dpkg --assert-support-predepends
+
+if [ "$1" = "upgrade" -a -f /etc/init.d/klogd ]; then
+    if [ -f /etc/init.d/klogd ]
+    then
+	set +e
+	sh /etc/init.d/klogd stop
+	set -e
+    fi
+fi
+
+exit 0
--- sysklogd-1.4.1/debian/prerm	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/prerm	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,8 @@
+#! /bin/sh
+
+set -e
+
+if [ \( "$1" = "upgrade" -o "$1" = "remove" \) -a -L /usr/doc/sysklogd ]
+then
+    rm -f /usr/doc/sysklogd
+fi
--- sysklogd-1.4.1/debian/prerm.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/prerm.klogd	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,8 @@
+#! /bin/sh
+
+set -e
+
+if [ \( "$1" = "upgrade" -o "$1" = "remove" \) -a -L /usr/doc/klogd ]
+then
+    rm -f /usr/doc/klogd
+fi
--- sysklogd-1.4.1/debian/rc	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/rc	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,38 @@
+#! /bin/sh
+# /etc/init.d/sysklogd: start the system log daemon.
+
+test -f /sbin/syslogd || exit 0
+
+# Options for start/restart the daemons
+#   For remote UDP logging use SYSLOGD="-r"
+#
+SYSLOGD=""
+
+case "$1" in
+  start)
+    echo -n "Starting system log daemon: syslogd"
+    start-stop-daemon --start --quiet --exec /sbin/syslogd -- $SYSLOGD
+    echo "."
+    ;;
+  stop)
+    echo -n "Stopping system log daemon: syslogd"
+    start-stop-daemon --stop --quiet --exec /sbin/syslogd --pidfile /var/run/syslogd.pid
+    echo "."
+    ;;
+  reload|force-reload)
+    start-stop-daemon --stop --quiet --signal 1 --exec /sbin/syslogd --pidfile /var/run/syslogd.pid
+    ;;
+  restart)
+    echo -n "Stopping system log daemon: syslogd"
+    start-stop-daemon --stop --quiet --exec /sbin/syslogd --pidfile /var/run/syslogd.pid
+    sleep 1
+    echo -n "Starting system log daemon: syslogd"
+    start-stop-daemon --start --quiet --exec /sbin/syslogd -- $SYSLOGD
+    echo "."
+    ;;
+  *)
+    echo "Usage: /etc/init.d/sysklogd {start|stop|reload|restart|force-reload}"
+    exit 1
+esac
+
+exit 0
--- sysklogd-1.4.1/debian/rc.klogd	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/rc.klogd	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,35 @@
+#! /bin/sh
+# /etc/init.d/klogd: start the kernel log daemon.
+
+test -f /sbin/klogd || exit 0
+
+#  Use KLOGD="-k /boot/System.map-$(uname -r)" to specify System.map
+#
+KLOGD=""
+
+case "$1" in
+  start)
+    echo -n "Starting kernel log daemon: klogd"
+    start-stop-daemon --start --quiet --exec /sbin/klogd -- $KLOGD
+    echo "."
+    ;;
+  stop)
+    echo -n "Stopping kernel log daemon: klogd"
+    start-stop-daemon --stop --quiet --exec /sbin/klogd --pidfile /var/run/klogd.pid
+    echo "."
+    ;;
+  restart|force-reload)
+    echo -n "Stopping kernel log daemon: klogd"
+    start-stop-daemon --stop --quiet --exec /sbin/klogd --pidfile /var/run/klogd.pid
+    echo "."
+    sleep 1
+    echo -n "Starting kernel log daemon: klogd"
+    start-stop-daemon --start --quiet --exec /sbin/klogd --exec /sbin/klogd -- $KLOGD
+    echo "."
+    ;;
+  *)
+    echo "Usage: /etc/init.d/klogd {start|stop|restart|force-reload}"
+    exit 1
+esac
+
+exit 0
--- sysklogd-1.4.1/debian/readme	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/readme	1998-12-28 18:51:09.000000000 +0200
@@ -0,0 +1,18 @@
+
+Additional information about system logging
+-------------------------------------------
+
+ . When logging into a fifo syslogd will stop loggin into this
+   logfile if no process is reading it and thus the buffer is filled
+   up.  Buffer size can be adjusted through the kernel.
+
+ . When using the Perl5 Sys::Syslog module (use Sys::Syslog;) to
+   provide logging to your Perl programs, please notice that since it
+   will open a UDP connection to the local syslogd you'll need to have
+   remote receiption enabled.
+
+   Joey Hess teaches us how to change this behaviour and switch to
+   using the Unix domain socket.  After the opening with "use" you
+   have to add "Sys::Syslog::setlogsock('unix');".
+
+   The default behaviour might change in future releases of Perl.
--- sysklogd-1.4.1/debian/rules	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/rules	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,156 @@
+#! /usr/bin/make -f
+
+#   Copyright 1994-98   joey@infodrom.north.de (Martin Schulze)
+#
+#   This program is free software; you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation; version 2 dated June, 1991.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program;  if not, write to the Free Software
+#   Foundation Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA
+SHELL=/bin/bash
+
+# The name and version of the source
+#
+source = $(shell grep "^Source: " debian/control|head -1|sed 's/Source: \(.*\)/\1/g')
+package = $(shell grep "^Package: " debian/control|head -1|sed 's/Package: \(.*\)/\1/g')
+version = $(shell grep "^$(source) " debian/changelog|head -1 |sed 's/.*(\(.*\)\-[^\-]*).*/\1/g')
+revision = $(shell grep "^$(source) " debian/changelog|head -1 |sed 's/.*([^\-]*\-\(.*\)).*/\1/g')
+
+installbin = install -g root -o root -m 755
+installdoc = install -g root -o root -m 644
+
+build:
+	$(MAKE) DEB="-DDEBRELEASE=\\\"$(revision)\\\""
+	pod2man --section=8 --lax --center="Debian GNU/Linux" \
+	  --release="Debian Project" debian/syslog-facility.pod \
+	  > syslog-facility.8
+	(cat README.linux; printf "\n\n"; cat README.1st; \
+	  printf "\n\nThe following important changes have been reported by the developers:\n\n"; \
+	  cat NEWS debian/readme) \
+	  > readme.txt
+	(cat debian/copyright; \
+	  printf "Syslogd stays under the following copyright:\n\n"; \
+	    sed -e '/^$$/,$$d' < syslogd.c; \
+	  printf "\nKlogd has the following copyright notice:\n\n"; \
+	    sed -e '/^$$/,$$d' < klogd.c ) \
+	  > copyright
+	touch stamp-build
+
+clean: debclean
+	rm -f stamp-build
+	$(MAKE) clobber
+	rm -rf *~ debian/*~ syslog-facility.8 readme.txt copyright
+
+debclean:
+# Cleans debian binary directories to allow binary creation
+	rm -rf debian/tmp.sysklogd debian/tmp.klogd
+	rm -f debian/{files,substvars}
+
+binary-indep:
+# Nothing to be done here
+
+binary-arch-sysklogd: debclean
+	test -f stamp-build || $(MAKE) -f debian/rules build
+	$(installbin) -d debian/tmp.sysklogd/DEBIAN
+	chown -R root.root debian/tmp.sysklogd
+	chmod -R g-ws debian/tmp.sysklogd
+	$(installbin) -d debian/tmp.sysklogd/usr/share/doc/$(package)
+	$(installbin) debian/{preinst,postinst,postrm,prerm} debian/tmp.sysklogd/DEBIAN/
+	$(installdoc) debian/conffiles debian/tmp.sysklogd/DEBIAN/
+	$(installdoc) debian/changelog debian/tmp.sysklogd/usr/share/doc/$(package)/changelog.Debian
+	#
+	$(installdoc) copyright debian/tmp.sysklogd/usr/share/doc/$(package)/
+	$(installdoc) readme.txt debian/tmp.sysklogd/usr/share/doc/$(package)/
+	$(installdoc) CHANGES debian/tmp.sysklogd/usr/share/doc/$(package)/changelog
+	gzip -9f debian/tmp.sysklogd/usr/share/doc/$(package)/{changelog.Debian,changelog,readme.txt}
+	#
+	$(installbin) -d debian/tmp.sysklogd/{usr/,}sbin
+	$(installbin) -s syslogd debian/tmp.sysklogd/sbin
+	$(installbin) debian/syslogd-listfiles debian/tmp.sysklogd/usr/sbin
+	$(installbin) debian/syslog-facility debian/tmp.sysklogd/usr/sbin
+	#
+	$(installbin) -d debian/tmp.sysklogd/etc/{init.d,cron.{daily,weekly}}
+	$(installbin) debian/rc debian/tmp.sysklogd/etc/init.d/sysklogd
+	$(installbin) debian/cron.weekly debian/tmp.sysklogd/etc/cron.weekly/sysklogd
+	$(installbin) debian/cron.daily debian/tmp.sysklogd/etc/cron.daily/sysklogd
+	$(installdoc) debian/syslog.conf debian/tmp.sysklogd/etc/syslog.conf
+	$(installbin) -d debian/tmp.sysklogd/var/log
+	#
+	$(installbin) -d debian/tmp.sysklogd/usr/share/man/man{5,8}
+	$(installdoc) syslog.conf.5 debian/tmp.sysklogd/usr/share/man/man5
+	$(installdoc) {sysklogd,syslogd,debian/syslogd-listfiles}.8 debian/tmp.sysklogd/usr/share/man/man8
+	$(installdoc) syslog-facility.8 debian/tmp.sysklogd/usr/share/man/man8
+	gzip -9 debian/tmp.sysklogd/usr/share/man/man?/*
+	#
+	dpkg-shlibdeps debian/tmp.sysklogd/sbin/syslogd
+	dpkg-gencontrol -psysklogd -Pdebian/tmp.sysklogd
+	dpkg --build debian/tmp.sysklogd ..
+
+binary-arch-klogd: debclean
+	test -f stamp-build || $(MAKE) -f debian/rules build
+	$(installbin) -d debian/tmp.klogd/DEBIAN
+	chown -R root.root debian/tmp.klogd
+	chmod -R g-ws debian/tmp.klogd
+	$(installbin) -d debian/tmp.klogd/usr/share/doc/klogd
+	$(installbin) debian/preinst.klogd debian/tmp.klogd/DEBIAN/preinst
+	$(installbin) debian/postinst.klogd debian/tmp.klogd/DEBIAN/postinst
+	$(installbin) debian/postrm.klogd debian/tmp.klogd/DEBIAN/postrm
+	$(installbin) debian/prerm.klogd debian/tmp.klogd/DEBIAN/prerm
+	$(installdoc) debian/conffiles.klogd debian/tmp.klogd/DEBIAN/conffiles
+	$(installdoc) debian/changelog debian/tmp.klogd/usr/share/doc/klogd/changelog.Debian
+	#
+	$(installdoc) copyright debian/tmp.klogd/usr/share/doc/klogd/
+	$(installdoc) readme.txt debian/tmp.klogd/usr/share/doc/klogd/
+	$(installdoc) CHANGES debian/tmp.klogd/usr/share/doc/klogd/changelog
+	gzip -9f debian/tmp.klogd/usr/share/doc/klogd/{changelog.Debian,changelog,readme.txt}
+	$(installdoc) modutils.patch debian/tmp.klogd/usr/share/doc/klogd
+	#
+	$(installbin) -d debian/tmp.klogd/sbin
+	$(installbin) -s klogd debian/tmp.klogd/sbin
+	#
+	$(installbin) -d debian/tmp.klogd/etc/init.d
+	$(installbin) debian/rc.klogd debian/tmp.klogd/etc/init.d/klogd
+	#
+	$(installbin) -d debian/tmp.klogd/usr/share/man/man8
+	$(installdoc) klogd.8 debian/tmp.klogd/usr/share/man/man8
+	gzip -9 debian/tmp.klogd/usr/share/man/man?/*
+	#
+	dpkg-shlibdeps debian/tmp.klogd/sbin/klogd
+	dpkg-gencontrol -pklogd -Pdebian/tmp.klogd
+	dpkg --build debian/tmp.klogd ..
+
+binary-arch: binary-arch-sysklogd binary-arch-klogd
+
+binary: binary-indep binary-arch
+
+source diff:
+	@echo >&2 'source and diff are obsolete - use dpkg-source -b' or dsc; false
+
+dsc:
+	-test -d debian/tmp.sysklogd -o -d debian/tmp.klogd \
+		&& $(MAKE) -f debian/rules clean
+	if [ ! -f ../$(source)_$(version).orig.tar.gz -a -f ../orig/$(source)_$(version).orig.tar.gz ]; \
+	then \
+	  ln -s orig/$(source)_$(version).orig.tar.gz ../$(source)_$(version).orig.tar.gz; \
+	  touch /tmp/stamp-$(source)-link; \
+	fi; \
+	cd .. && dpkg-source -b $(source)-$(version)
+	if [ -f /tmp/stamp-$(source)-link ]; then \
+	  rm ../$(source)_$(version).orig.tar.gz /tmp/stamp-$(source)-link; \
+	fi
+
+checkroot:
+	$(checkdir)
+	test root = "`whoami`"
+
+dist: binary dsc
+
+.PHONY: binary binary-arch binary-indep clean checkroot
+
--- sysklogd-1.4.1/debian/syslog-facility	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/syslog-facility	1998-10-12 23:28:38.000000000 +0300
@@ -0,0 +1,140 @@
+#! /usr/bin/perl -w
+
+# Copyright 1998 Hertzog Raphaël
+# You can use this script under the term of the GPL v2 or later.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111, USA.
+
+my $conf_file = '/etc/syslog.conf';
+
+## BUGS :
+#  . This script doesn't know about multi-lines configuration (ie with '\')
+#  . With a line like that "mail,local0.* /anything" 
+#    "syslog-facility remove local0" would remove the entire line
+#    => should not be a problem since lines installed by this script
+#    cannot use such syntax
+##
+
+## Nothing to modify after this line ##
+
+my $command = lc(shift);
+
+usage() if ($command !~ /^(?:set|remove)$/); 
+usage() if (not scalar(@ARGV));
+
+if ($command eq "set")
+{
+    usage() if (int(scalar(@ARGV) / 2) != scalar(@ARGV) / 2);
+    # find a free localx facility
+    my $facility = get_first_free_facility();
+    # if none stop immediately
+    if ($facility eq "none") {
+	print "none\n";
+	exit 1;
+    }
+    # ok append the lines asked
+    open (CONF, ">>$conf_file") || 
+        die "Can't open $conf_file in write mode: $!\n";
+    my ($pri,$file,$line);
+    while (defined($pri = shift)) {
+	$file = shift;
+	$line = "";
+	foreach (split(/;/,$pri)) {
+	    $_ =~ s/all/*/g;
+	    $line .= ";" if ($line);
+	    $line .= "$facility.$_";
+	}
+	$line .= "\t\t$file\n";
+	print CONF $line;
+    }
+    close CONF;
+    print "$facility\n";
+    exit 0;
+
+} elsif ($command eq "remove") {
+
+    my $facility = lc(shift);
+    my ($left,$file,$line);
+    open (CONF, $conf_file) || die "Can't open $conf_file: $!\n";
+    open (CONFNEW, ">$conf_file.new")  || 
+	die "Can't open $conf_file.new in write mode: $!\n";
+    while (defined($_=<CONF>)) {
+	# Write all "simple" lines like empty lines and comments
+	if (/^\s*$/ or /^\s*#/ or /\\$/) {
+	    print CONFNEW $_;
+	    next;
+	}
+	# Otherwise look if the facility to remove appears in the line
+	if (/^\s*(\S+)\s+(\S+)\s*/) {
+	    $left = $1; $file = $2; chomp $file;
+	    # It doesn't appers => write
+	    if ($left !~ /$facility/i) {
+		print CONFNEW $_;
+		next;
+	    }
+	    # It appears => write a new line without the localx facility
+	    $line = "";
+	    foreach (split(/;/,$left)) {
+		if (not /$facility/i) {
+		    $line .= ";" if ($line);
+		    $line .= $_;
+		}
+	    }
+	    next if ($line eq "");
+	    $line .= "\t\t$file\n";
+	    print CONFNEW $line;
+	}
+    }
+    close CONFNEW;
+    close CONF;
+    rename ("$conf_file.new", "$conf_file");
+}
+
+sub get_first_free_facility {
+
+    my @facility = (0) x 8;
+    my ($left,$fac);
+    open(CONF, $conf_file) || die "Can't open $conf_file: $!\n";
+    while(defined($_=<CONF>))
+    {
+	next if (/^\s*$/);
+	next if (/^\s*#/);
+        next if (/\\$/);
+	next if (not /^\s*(\S+)\s+(\S+)\s*$/);
+	$left = $1;
+	foreach $fac (split(/;/,$left)) {
+	    $facility[$1]++ if ($fac =~ /local(\d)/i);
+	}
+    }
+    foreach $fac (0..7)	{
+	return "local$fac" if ($facility[$fac] == 0);
+    }
+    return "none";
+}
+
+sub usage {
+
+    die  "syslog-facility - Copyright (c) 1998 Hertzog Raphaël\n"
+	."Usage : $0 set <set_of_priority> <logfile> ... \n"
+	."        it returns the 'LOCALx' string you have the right to use.\n"
+	."        $0 remove <facility>\n"
+	."Example: $0 set all /var/log/all\n"
+	."         $0 set all\\;\\!=info /var/log/all-without-info\n"
+	."         $0 set =err /var/log/errors =warning /var/log/warn\n"
+	."         $0 remove LOCAL1\n";   
+}
+
+
+
--- sysklogd-1.4.1/debian/syslog-facility.pod	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/syslog-facility.pod	1998-10-13 01:01:38.000000000 +0300
@@ -0,0 +1,58 @@
+=head1 NAME
+
+syslog-facility - Setup and remove LOCALx facility for sysklogd
+
+=head1 SYNOPSIS
+
+B<syslog-facility> B<set> I<set_of_priority> I<log_file>  ...
+
+B<syslog-facility> B<remove> I<facility>
+
+=head1 DESCRIPTION
+
+B<syslog-facility> can be used to setup a syslog facility and to remove
+it. The primary use of this perl script is to allow packages to
+automatically setup (and remove) a LOCALx facility. With the
+first argument beeing I<set>, it will output a string containing
+the first free LOCALx facility that has been assigned to your request :
+
+  # syslog-facility set all /var/log/sympa
+  local0
+
+This does mean that a line "local0.*	/var/log/sympa" has been
+added to the syslog.conf file. If no free LOCALx facility has been
+found, then it outputs "none" (with a end of line).
+
+You can use different set of priorities (most of what syslogd supports) :
+
+  # syslog-facility set 'all;!=debug;' /var/log/sympa '=debug' /var/log/sympa-d
+  local1
+
+Pay particular attention to shell escapes since ';','!' have special
+meanings for them. The syntax is similar to syslog.conf except that
+'*' must be replaced by 'all'.
+
+In that case you will have two lines added two the syslog.conf file :
+
+  local1.*;local1.!=debug	/var/log/sympa
+  local1.=debug			/var/log/sympa-d
+
+If the first argument is I<remove>, then it will remove all references
+to a precise LOCALx facility. Example :
+
+  # syslog-facility remove local1
+
+It will remove all references to the local1 facility. 
+
+=head1 BUGS
+
+The script doesn't know about multi-lines configuration (ie with '\').
+
+With a line like that "mail,local0.* /anything", "syslog-facility remove
+local0" would remove the entire line. It should not be a problem since
+lines installed by this script cannot use this syntax.
+
+=head1 SEE ALSO
+
+B<syslog.conf>(5), B<sysklogd>(8).
+ 
--- sysklogd-1.4.1/debian/syslog.conf	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/syslog.conf	1998-10-12 19:59:31.000000000 +0300
@@ -0,0 +1,71 @@
+#  /etc/syslog.conf	Configuration file for syslogd.
+#
+#			For more information see syslog.conf(5)
+#			manpage.
+
+#
+# First some standard logfiles.  Log by facility.
+#
+
+auth,authpriv.*			/var/log/auth.log
+*.*;auth,authpriv.none		-/var/log/syslog
+#cron.*				/var/log/cron.log
+daemon.*			-/var/log/daemon.log
+kern.*				-/var/log/kern.log
+lpr.*				-/var/log/lpr.log
+mail.*				/var/log/mail.log
+user.*				-/var/log/user.log
+uucp.*				-/var/log/uucp.log
+
+#
+# Logging for the mail system. Split it up so that
+# it is easy to write scripts to parse these files.
+#
+mail.info			-/var/log/mail.info
+mail.warn			-/var/log/mail.warn
+mail.err			/var/log/mail.err
+
+# Logging for INN news system
+#
+news.crit			/var/log/news/news.crit
+news.err			/var/log/news/news.err
+news.notice			-/var/log/news/news.notice
+
+#
+# Some `catch-all' logfiles.
+#
+*.=debug;\
+	auth,authpriv.none;\
+	news.none;mail.none	-/var/log/debug
+*.=info;*.=notice;*.=warn;\
+	auth,authpriv.none;\
+	cron,daemon.none;\
+	mail,news.none		-/var/log/messages
+
+#
+# Emergencies are sent to everybody logged in.
+#
+*.emerg				*
+
+#
+# I like to have messages displayed on the console, but only on a virtual
+# console I usually leave idle.
+#
+#daemon,mail.*;\
+#	news.=crit;news.=err;news.=notice;\
+#	*.=debug;*.=info;\
+#	*.=notice;*.=warn	/dev/tty8
+
+# The named pipe /dev/xconsole is for the `xconsole' utility.  To use it,
+# you must invoke `xconsole' with the `-file' option:
+# 
+#    $ xconsole -file /dev/xconsole [...]
+#
+# NOTE: adjust the list below, or you'll go crazy if you have a reasonably
+#      busy site..
+#
+daemon.*;mail.*;\
+	news.crit;news.err;news.notice;\
+	*.=debug;*.=info;\
+	*.=notice;*.=warn	|/dev/xconsole
+
--- sysklogd-1.4.1/debian/syslogd-listfiles	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/syslogd-listfiles	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,113 @@
+#! /usr/bin/perl
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111, USA.
+
+# Tue Jan 13 01:47:57 MET 1998: Martin Schulze <joey@infodrom.north.de>
+#	Fixed typo.
+#
+#	Modified logfile detection routine to take care of double
+#	listed logfiles.  It does this by using a hash instead of
+#	immediate printout.  Thanks to Roman Hodek <roman@debian.org>
+#	for providing an appropriate patch.
+
+$conf = "/etc/syslog.conf";
+$opt_daily = 1;
+$opt_all = 0;
+$opt_auth = 0;
+$opt_ign_size = 0;
+$opt_news = 0;
+$opt_skip = 0;
+
+sub usage
+{
+    print STDERR
+"
+Debian GNU/Linux syslogd-listfiles =VER=.  Copyright (C) 1997
+Martin Schulze.  This is free software; see the GNU General Public Licence
+version 2 or later for copying conditions.  There is NO warranty.
+
+Usage: syslogd-listfiles <options>
+Options: -f file	specifies another syslog.conf file
+         -a | --all	list all files (including news)
+         --auth		list all files containing auth.<some prio>
+	 --ignore-size  don't rotate files which got too large (default: 2MB)
+         --news		include news logfiles, too
+         -w | --weekly	use weekly pattern instead of daily
+         -s pattern	skip files matching pattern
+";
+}
+
+while (@ARGV) {
+    $_=shift(@ARGV);
+    if (m/^-f$/) {
+	$conf = shift(@ARGV);
+    } elsif (m/^-s$/) {
+	$opt_skip = shift(@ARGV);
+    } elsif (m/^(--weekly|-w)$/) {
+	$opt_daily = 0;
+    } elsif (m/^(-a|--all)$/) {
+	$opt_all = 1;
+    } elsif (m/^--auth$/) {
+	$opt_auth = 1;
+    } elsif (m/^--ignore-size/) {
+	$opt_ign_size = 1;
+    } elsif (m/^--news$/) {
+	$opt_news = 1;
+    } else {
+	&usage();exit (0);
+    }
+}
+
+open (C, $conf) || die "Can't open $conf, $!";
+while (<C>) {
+    next if (/^(\#|$)/);
+    chop;
+
+    s/\s*(\S.*)$/$1/ if ($line);
+
+    $line .= $_;
+    chop ($line) if (/\\$/);
+    if (!/\\$/) {
+	$line =~ s/\s+/\t/;
+	$line =~ s/\t-/\t/;
+	push (@lines, $line) if ($line =~ /\t\/(?!dev\/)/);
+	$line = "";
+    }
+}
+close (C);
+
+foreach $line (@lines) {
+    ($pat,$file) = split (/\t/,$line);
+
+    # handled by news.daily from INN
+    next if (!$opt_news && ($pat =~ /news\.(crit|err|notice)/));
+
+    if ($opt_all) {
+	$output{$file} = 1;
+    } elsif ($opt_auth) {
+	$output{$file} = 1 if ($pat =~ /auth[^\.]*\.(?!none).*/);
+    } else {
+	$i = ($pat =~ /\*\.\*/);
+	$output{$file} = 1 if (($i && $opt_daily) || (!$i && !$opt_daily)
+			       || (!$opt_ign_size && (stat $file)[7] >= 2097152 && $opt_daily)); # 2MB
+    }
+}
+
+foreach $file (keys (%output)) {
+    $skip = $file;
+    if ($skip !~ /$opt_skip/) {
+	printf "%s\n", $file;
+    }
+}
--- sysklogd-1.4.1/debian/syslogd-listfiles.8	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/syslogd-listfiles.8	2001-03-12 01:00:53.000000000 +0200
@@ -0,0 +1,83 @@
+.\" syslogd-listfiles.8 - generic routine for /etc/syslog.conf
+.\" Copyright (c) 1997  Martin Schulze <joey@debian.org>
+.\" 
+.\" This program is free software; you can redistribute it and/or modify
+.\" it under the terms of the GNU General Public License as published by
+.\" the Free Software Foundation; either version 2 of the License, or
+.\" (at your option) any later version.
+.\" 
+.\" This program is distributed in the hope that it will be useful,
+.\" but WITHOUT ANY WARRANTY; without even the implied warranty of
+.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+.\" GNU General Public License for more details.
+.\" 
+.\" You should have received a copy of the GNU General Public License
+.\" along with this program; if not, write to the Free Software
+.\" Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+.\"
+.TH SYSLOGD-LISTFILES 8 "1 January 1998" "Debian project" "Debian GNU/Linux"
+.SH NAME
+syslogd-listfiles \- list system logfiles
+.SH SYNOPSIS
+.B syslogd-listfiles
+.RB [ " \-a " | " \-\-all " ]
+.RB [ " \-\-auth " ] 
+.RB [ " \-f "
+.I config file
+]
+.RB [ " \-s "
+.I pattern
+]
+.RB [ " \-\-news " ]
+.RB [ " \-w " | " \-\-weekly " ]
+.SH DESCRIPTION
+.B Sysklogd
+provides a modern mechanism to log system messages.  This mechanism is
+controlled by the file
+.IR /etc/syslog.conf .
+Messages are divided into logfiles which can grow quite big.
+Therefore the logfiles need to be rotated once per day, per week or
+per month.  On a Debian GNU/Linux system most of the files are rotated
+once per week.
+
+This program helps the rotation script to determine which logfiles
+need to be rotated and when.  It takes care of news logfiles that are
+handled by the
+.B news.daily
+script from INN.  In conjunction with the scripts in
+.IR /etc/cron.daily " and " /etc/cron.weekly
+it takes care of files with sensitive information.
+
+By default a list of files for daily rotation is generated.  At the
+moment this option only reflects entries that contain "*.*" as
+facility.priority.
+
+.SH OPTIONS
+.TP
+.BR "\-a" ", " "\-\-all"
+List all logfiles and ignore all other rules.
+.TP
+.BI "\-f " "config file"
+Specify an alternative configuration file instead of
+.IR /etc/syslog.conf ","
+which is the default.
+.TP
+.BI "\-p " "pattern"
+Specify a regular expression for files that must not be listed.
+.TP
+.B "\-\-auth"
+Only list files containing sensitive information such as the one
+containing auth.*.
+.TP
+.B "\-\-news"
+Don't exclude news logfiles which are normally handled by the
+.B news.daily
+script from INN.
+.TP
+.BR "\-w" ", " "\-\-weekly"
+Generate the list for weekly rotation.
+.LP
+.SH SEE ALSO
+.BR syslog.conf (5),
+.BR syslogd (8),
+.BR savelog (8).
--- sysklogd-1.4.1/debian/technical	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/debian/technical	1997-06-11 00:08:04.000000000 +0300
@@ -0,0 +1,10 @@
+  Some technical information - unsorted
+  -------------------------------------
+
+  1. Logfiles in syslogd
+
+     As of version 1.3 syslogd doesn't come with a static array of
+     logfiles.  Although the logfiles are still organized in an array.
+     This array is initialized when the configuration file (default is
+     /etc/syslog.conf) is read.  Whenever a new line is read the
+     length of the array it prolongued.
--- sysklogd-1.4.1/klogd.8	2001-03-11 21:35:51.000000000 +0200
+++ sysklogd/klogd.8	2005-03-11 18:12:09.000000000 +0200
@@ -3,8 +3,9 @@
 .\" Sun Jul 30 01:35:55 MET: Martin Schulze: Updates
 .\" Sun Nov 19 23:22:21 MET: Martin Schulze: Updates
 .\" Mon Aug 19 09:42:08 CDT 1996: Dr. G.W. Wettstein: Updates
+.\" Fri Mar 11 17:11:46 CET 2005: Martin Schulze: sysctl updates
 .\"
-.TH KLOGD 8 "21 August, 1999" "Version 1.4" "Linux System Administration"
+.TH KLOGD 8 "11 March 2005" "Version 1.4" "Linux System Administration"
 .SH NAME
 klogd \- Kernel Log Daemon
 .LP
@@ -38,6 +39,12 @@
 .TP
 .BI "\-c " n
 Sets the default log level of console messages to \fIn\fR.
+This should nowardays be configured with the
+.BR sysctl (8)
+program, e.g. via
+.BI "sysctl kernel.printk=" n
+or via
+.IR /etc/sysctl.conf .
 .TP
 .B "\-d"
 Enable debugging mode.  This will generate \fBLOTS\fR of output to
@@ -321,7 +328,7 @@
 .B klogd
 to reload the module symbol information whenever a protection fault
 is detected.  Caution should be used before invoking the program in
-\'paranoid\' mode.  The stability of the kernel and the operating
+\&'paranoid\&' mode.  The stability of the kernel and the operating
 environment is always under question when a protection fault occurs.
 Since the klogd daemon must execute system calls in order to read the
 module symbol information there is the possibility that the system may
--- sysklogd-1.4.1/klogd.c	2001-03-11 21:40:10.000000000 +0200
+++ sysklogd/klogd.c	2004-04-29 16:29:03.000000000 +0300
@@ -243,6 +243,9 @@
  *	people have submitted patches: Troels Walsted Hansen
  *	<troels@thule.no>, Wolfgang Oertl <Wolfgang.Oertl@uibk.ac.at>
  *	and Thomas Roessler.
+ * Thu Apr 29 15:24:07 2004: Solar Designer <solar@openwall.com>
+ *	Prevent potential buffer overflow in reading messages from the
+ *	kernel log rinbuffer.
  */
 
 
@@ -938,7 +941,7 @@
 	 * messages into this fresh buffer.
 	 */
 	memset(log_buffer, '\0', sizeof(log_buffer));
-	if ( (rdcnt = ksyslog(2, log_buffer, sizeof(log_buffer))) < 0 )
+	if ( (rdcnt = ksyslog(2, log_buffer, sizeof(log_buffer)-1)) < 0 )
 	{
 		if ( errno == EINTR )
 			return;
--- sysklogd-1.4.1/ksym.c	2000-09-12 23:53:31.000000000 +0200
+++ sysklogd/ksym.c	2004-07-16 09:48:27.000000000 +0300
@@ -105,6 +105,15 @@
  *
  * Tue Sep 12 23:48:12 CEST 2000: Martin Schulze <joey@infodrom.ffis.de>
  *	Close symbol file in InitKsyms() when an error occurred.
+ *
+ * Thu Apr 29 18:07:16 CEST 2004: Dmitry Levin <ldv@altlinux.org>
+ *	Close file descriptor in FindSymbolFile() in order not to leak
+ *	file descriptors.
+ *
+ * Fri Jul 16 08:32:49 CEST 2004: Ulf Härnhammar <Ulf.Harnhammar.9485@student.uu.se>
+ *	Added boundary check for fscanf() in InitKsyms() and
+ *	CheckMapVersion() to prevent an unintended crash when reading
+ *	an incorrect System.map.
  */
 
 
@@ -236,7 +245,7 @@
 	 */
 	while ( !feof(sym_file) )
 	{
-		if ( fscanf(sym_file, "%lx %c %s\n", &address, &type, sym)
+		if ( fscanf(sym_file, "%lx %c %511s\n", &address, &type, sym)
 		    != 3 )
 		{
 			Syslog(LOG_ERR, "Error in symbol table input (#1).");
@@ -344,6 +353,7 @@
 		if ( (sym_file = fopen(symfile, "r")) != (FILE *) 0 ) {
 			if (CheckMapVersion(symfile) == 1)
 				file = symfile;
+			fclose (sym_file);
 		}
 		if (sym_file == (FILE *) 0 || file == (char *) 0) {
 			sprintf (symfile, "%s", *mf);
@@ -352,6 +362,7 @@
 			if ( (sym_file = fopen(symfile, "r")) != (FILE *) 0 ) {
 				if (CheckMapVersion(symfile) == 1)
 					file = symfile;
+				fclose (sym_file);
 			}
 		}
 
@@ -533,7 +544,7 @@
 		version = 0;
 		while ( !feof(sym_file) && (version == 0) )
 		{
-			if ( fscanf(sym_file, "%lx %c %s\n", &address, \
+			if ( fscanf(sym_file, "%lx %c %511s\n", &address, \
 				    &type, sym) != 3 )
 			{
 				Syslog(LOG_ERR, "Error in symbol table input (#2).");
@@ -899,3 +910,11 @@
 	return;
 }
 #endif
+
+/*
+ * Local variables:
+ *  c-indent-level: 8
+ *  c-basic-offset: 8
+ *  tab-width: 8
+ * End:
+ */
--- sysklogd-1.4.1/ksym_mod.c	2000-09-12 23:15:28.000000000 +0200
+++ sysklogd/ksym_mod.c	2004-03-31 18:47:08.000000000 +0300
@@ -78,6 +78,11 @@
  *
  * Tue Sep 12 23:11:13 CEST 2000: Martin Schulze <joey@infodrom.ffis.de>
  *	Changed llseek() to lseek64() in order to skip a libc warning.
+ *
+ * Wed Mar 31 17:35:01 CEST 2004: Martin Schulze <joey@infodrom.org>
+ *	Removed references to <linux/module.h> since it doesn't work
+ *	anymore with its recent content from Linux 2.4/2.6, created
+ *	module.h locally instead.
  */
 
 
@@ -89,11 +94,12 @@
 #include <errno.h>
 #include <sys/fcntl.h>
 #include <sys/stat.h>
+#include "module.h"
 #if !defined(__GLIBC__)
 #include <linux/time.h>
-#include <linux/module.h>
+#include <linux/linkage.h>
 #else /* __GLIBC__ */
-#include <linux/module.h>
+#include <linux/linkage.h>
 extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
 extern int get_kernel_syms __P ((struct kernel_sym *__table));
 #endif /* __GLIBC__ */
@@ -107,7 +113,7 @@
 
 #if !defined(__GLIBC__)
 /*
- * The following bit uses some kernel/library magic to product what
+ * The following bit uses some kernel/library magic to produce what
  * looks like a function call to user level code.  This function is
  * actually a system call in disguise.  The purpose of the getsyms
  * call is to return a current copy of the in-kernel symbol table.
--- sysklogd-1.4.1/module.h	1970-01-01 03:00:00.000000000 +0300
+++ sysklogd/module.h	2006-05-25 10:15:57.000000000 +0300
@@ -0,0 +1,90 @@
+/*
+    module.h - Miscellaneous module definitions
+    Copyright (c) 1996 Richard Henderson <rth@tamu.edu>
+    Copyright (c) 2004 Martin Schulze <joey@infodrom.org>
+
+    This file is part of the sysklogd package.
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+*/
+
+/* ChangeLog:
+ *
+ * Wed Mar 31 17:35:01 CEST 2004: Martin Schulze <joey@infodrom.org>
+ *	Created local copy of module.h based on the content of Linux
+ *	2.2 since <linux/module.h> doesn't work anymore with its
+ *	recent content from Linux 2.4/2.6.
+ * Thu May 25 09:14:33 CEST 2006: Martin Schulze <joey@infodrom.org>
+ *	Removed asm/atomic.h since it is not needed anymore.
+ */
+
+#define MODULE_NAME_LEN 60
+
+struct kernel_sym
+{
+	unsigned long value;
+	char name[MODULE_NAME_LEN];
+};
+
+
+struct list_head {
+	struct list_head *next, *prev;
+};
+
+
+struct module_info
+{
+	unsigned long addr;
+	unsigned long size;
+	unsigned long flags;
+	long usecount;
+};
+
+
+struct module
+{
+	unsigned long size_of_struct;	/* == sizeof(module) */
+	struct module *next;
+	const char *name;
+	unsigned long size;
+
+	union
+	{
+		int usecount;
+		long pad;
+	} uc;				/* Needs to keep its size - so says rth */
+
+	unsigned long flags;		/* AUTOCLEAN et al */
+
+	unsigned nsyms;
+	unsigned ndeps;
+
+	struct module_symbol *syms;
+	struct module_ref *deps;
+	struct module_ref *refs;
+	int (*init)(void);
+	void (*cleanup)(void);
+	const struct exception_table_entry *ex_table_start;
+	const struct exception_table_entry *ex_table_end;
+#ifdef __alpha__
+	unsigned long gp;
+#endif
+	/* Members past this point are extensions to the basic
+	   module support and are optional.  Use mod_opt_member()
+	   to examine them.  */
+	const struct module_persist *persist_start;
+	const struct module_persist *persist_end;
+	int (*can_unload)(void);
+};
--- sysklogd-1.4.1/pidfile.c	1998-02-11 00:37:12.000000000 +0200
+++ sysklogd/pidfile.c	2003-09-28 05:38:18.000000000 +0300
@@ -87,7 +87,7 @@
   int fd;
   int pid;
 
-  if ( ((fd = open(pidfile, O_RDWR|O_CREAT, 0644)) == -1)
+  if ( ((fd = open(pidfile, O_RDWR|O_CREAT|O_TRUNC, 0644)) == -1)
        || ((f = fdopen(fd, "r+")) == NULL) ) {
       fprintf(stderr, "Can't open or create %s.\n", pidfile);
       return 0;
--- sysklogd-1.4.1/sysklogd.8	2001-03-11 21:35:51.000000000 +0200
+++ sysklogd/sysklogd.8	2004-07-09 20:33:32.000000000 +0300
@@ -84,7 +84,7 @@
 .B MAXFUNIX
 within the syslogd.c source file.  An example for a chroot() daemon is
 described by the people from OpenBSD at
-http://www.psionic.com/papers/dns.html.
+<http://www.guides.sk/psionic/dns/>.
 .TP
 .B "\-d"
 Turns on debug mode.  Using this the daemon will not proceed a 
@@ -117,7 +117,8 @@
 between two \fI-- MARK --\fR lines is 20 minutes.  This can be changed
 with this option.  Setting the
 .I interval
-to zero turns it off entirely.
+to zero turns it off entirely.  Depending on other log messages
+generated these lines may not be written consecutively.
 .TP
 .B "\-n"
 Avoid auto-backgrounding.  This is needed especially if the
@@ -364,8 +365,10 @@
 
 To avoid this in further times no messages that were received from a
 remote host are sent out to another (or the same) remote host
-anymore.  If there are scenarios where this doesn't make sense, please
-drop me (Joey) a line.
+anymore.  If you experience are setup in which this doesn't make
+sense, please use the
+.B \-h
+commandline switch.
 
 If the remote host is located in the same domain as the host, 
 .B syslogd
--- sysklogd-1.4.1/syslog.c	2001-03-11 21:35:51.000000000 +0200
+++ sysklogd/syslog.c	2003-08-27 18:56:01.000000000 +0300
@@ -47,6 +47,9 @@
  * Sun Mar 11 20:23:44 CET 2001: Martin Schulze <joey@infodrom.ffis.de>
  *	Use SOCK_DGRAM for loggin, renables it to work.	
  *
+ * Wed Aug 27 17:48:16 CEST 2003: Martin Schulze <joey@Infodrom.org>
+ *	Improved patch by Michael Pomraning <mjp@securepipe.com> to
+ *	reconnect klogd to the logger after it went away.
  */
 
 #include <sys/types.h>
@@ -98,6 +101,7 @@
 	register char *p;
 	time_t now;
 	int fd, saved_errno;
+	int result;
 	char tbuf[2048], fmt_cpy[1024], *stdp = (char *) 0;
 
 	saved_errno = errno;
@@ -167,7 +171,16 @@
 	}
 
 	/* output the message to the local logger */
-	if (write(LogFile, tbuf, cnt + 1) >= 0 || !(LogStat&LOG_CONS))
+	result = write(LogFile, tbuf, cnt + 1);
+
+	if (result == -1
+	    && (errno == ECONNRESET || errno == ENOTCONN || errno == ECONNREFUSED)) {
+		closelog();
+		openlog(LogTag, LogStat | LOG_NDELAY, LogFacility);
+		result = write(LogFile, tbuf, cnt + 1);
+	}
+
+	if (result >= 0 || !(LogStat&LOG_CONS))
 		return;
 
 	/*
--- sysklogd-1.4.1/syslog.conf.5	1999-08-21 13:49:14.000000000 +0300
+++ sysklogd/syslog.conf.5	2003-05-22 22:31:20.000000000 +0300
@@ -64,7 +64,7 @@
 The
 .I facility
 is one of the following keywords:
-.BR auth ", " authpriv ", " cron ", " daemon ", " kern ", " lpr ", "
+.BR auth ", " authpriv ", " cron ", " daemon ", " ftp ", " kern ", " lpr ", "
 .BR mail ", " mark ", " news ", " security " (same as " auth "), "
 .BR syslog ", " user ", " uucp " and " local0 " through " local7 .
 The keyword 
@@ -121,12 +121,21 @@
 
 This 
 .BR syslogd (8)
-has a syntax extension to the original BSD source, that makes its use
+has a syntax extension to the original BSD source, which makes its use
 more intuitively.  You may precede every priority with an equation sign
-(``='') to specify only this single priority and not any of the
-above.  You may also (both is valid, too) precede the priority with an
-exclamation mark (``!'') to ignore all that priorities, either exact
-this one or this and any higher priority.  If you use both extensions
+(``='') to specify that
+.B syslogd
+should only refer to this single priority and not this priority and
+all higher priorities.
+
+You may also precide the priority with an exclamation mark (``!'') if
+you want
+.B syslogd
+to ignore this priority and all higher priorities.
+You may even use both, the exclamation mark and the equation sign if
+you want
+.B syslogd
+to ignore only this single priority.  If you use both extensions
 than the exclamation mark must occur before the equation sign, just
 use it intuitively.
 
@@ -300,7 +309,7 @@
 .B syslogd
 log all messages that come with either the
 .BR info " or the " notice
-facility into the file
+priority into the file
 .IR /var/log/messages ,
 except for all messages that use the
 .B mail
--- sysklogd-1.4.1/syslogd.c	2001-03-11 21:40:10.000000000 +0200
+++ sysklogd/syslogd.c	2006-05-25 10:49:54.000000000 +0300
@@ -441,6 +441,45 @@
  *	Don't return a closed fd if `-a' is called with a wrong path.
  *	Thanks to Bill Nottingham <notting@redhat.com> for providing
  *	a patch.
+ * Thu Apr 13 05:08:10 CEST 2001: Jon Burgess <Jon_Burgess@eur.3com.com>
+ *	Moved the installation of the signal handler up a little bit
+ *	so it guaranteed to be available when the child is forked,
+ *	hence, fixing a  race condition.  This used to create problems
+ *	with UML and fast machines.
+ *
+ * Sat Apr 17 18:03:05 CEST 2004: Steve Grubb <linux_4ever@yahoo.com>
+ *	Correct memory allocation for for commandline arguments in
+ *	crunch_list().
+ *
+ * Thu Apr 29 12:38:39 CEST 2004: Solar Designer <solar@openwall.com>
+ *	Applied Openwall paranoia patches to improve crunch_list().
+ *
+ * Tue May  4 16:47:30 CEST 2004: Solar Designer <solar@openwall.com>
+ *	Ensure that "len" is not placed in a register, and that the
+ *	endtty() signal handler is not installed too early which could
+ *	cause a segmentation fault or worse.
+ *
+ * Tue May  4 16:52:01 CEST 2004: Solar Designer <solar@openwall.com>
+ *	Adjust the size of a variable to prevent a buffer overflow
+ *	should _PATH_DEV ever contain something different than "/dev/".
+ *
+ * Tue Nov  2 20:28:23 CET 2004: Colin Phipps <cph@cph.demon.co.uk>
+ *	Don't block on the network socket, in case a packet gets lost
+ *	between select and recv.
+ *
+ * Sun Nov  7 12:28:47 CET 2004: Martin Schulze <joey@infodrom.org>
+ *	Discard any timestamp information found in received syslog
+ *	messages.  This will affect local messages sent from a
+ *	different timezone.
+ *
+ * Sun Nov  7 13:47:00 CET 2004: Martin Schulze <joey@infodrom.org>
+ *	Remove trailing newline when forwarding messages.
+ *
+ * Thu May 25 09:47:38 CEST 2006: Martin Schulze <joey@infodrom.org>
+ *	Reset the 'restart' flag immediately after entering the
+ *	restart code, so that subsequent SIGHUPs are able to set it
+ *	again and cause a new restart.  This fixes a race condition
+ *	when somebody sends tons of HUP signals.
  */
 
 
@@ -890,11 +929,11 @@
 		dprintf("Checking pidfile.\n");
 		if (!check_pid(PidFile))
 		{
+			signal (SIGTERM, doexit);
 			if (fork()) {
 				/*
 				 * Parent process
 				 */
-				signal (SIGTERM, doexit);
 				sleep(300);
 				/*
 				 * Not reached unless something major went wrong.  5
@@ -992,6 +1031,7 @@
 	(void) signal(SIGCHLD, reapchild);
 	(void) signal(SIGALRM, domark);
 	(void) signal(SIGUSR1, Debug ? debug_switch : SIG_IGN);
+	(void) signal(SIGXFSZ, SIG_IGN);
 	(void) alarm(TIMERINTVL);
 
 	/* Create a partial message table for all file descriptors. */
@@ -1074,9 +1114,9 @@
 				  (fd_set *) NULL, (struct timeval *) NULL);
 		if ( restart )
 		{
+			restart = 0;
 			dprintf("\nReceived SIGHUP, reloading syslogd.\n");
 			init();
-			restart = 0;
 			continue;
 		}
 		if (nfds == 0) {
@@ -1141,13 +1181,13 @@
 				 */
 				printchopped(from, line, \
  					     i + 2,  finet);
-			} else if (i < 0 && errno != EINTR) {
+			} else if (i < 0 && errno != EINTR && errno != EAGAIN) {
 				dprintf("INET socket error: %d = %s.\n", \
 					errno, strerror(errno));
 				logerror("recvfrom inet");
 				/* should be harmless now that we set
 				 * BSDCOMPAT on the socket */
-				sleep(10);
+				sleep(1);
 			}
 		}
 #endif
@@ -1216,6 +1256,7 @@
 {
 	int fd, on = 1;
 	struct sockaddr_in sin;
+	int sockflags;
 
 	fd = socket(AF_INET, SOCK_DGRAM, 0);
 	if (fd < 0) {
@@ -1241,6 +1282,24 @@
 		close(fd);
 		return -1;
 	}
+	/* We must not block on the network socket, in case a packet
+	 * gets lost between select and recv, otherise the process
+	 * will stall until the timeout, and other processes trying to
+	 * log will also stall.
+	 */
+	if ((sockflags = fcntl(fd, F_GETFL)) != -1) {
+		sockflags |= O_NONBLOCK;
+		/*
+		 * SETFL could fail too, so get it caught by the subsequent
+		 * error check.
+		 */
+		sockflags = fcntl(fd, F_SETFL, sockflags);
+	}
+	if (sockflags == -1) {
+		logerror("fcntl(O_NONBLOCK), suspending inet");
+		close(fd);
+		return -1;
+	}
 	if (bind(fd, (struct sockaddr *) &sin, sizeof(sin)) < 0) {
 		logerror("bind, suspending inet");
 		close(fd);
@@ -1254,30 +1313,26 @@
 crunch_list(list)
 	char *list;
 {
-	int count, i;
+	int i, m, n;
 	char *p, *q;
 	char **result = NULL;
 
 	p = list;
 	
 	/* strip off trailing delimiters */
-	while (p[strlen(p)-1] == LIST_DELIMITER) {
-		count--;
+	while (*p && p[strlen(p)-1] == LIST_DELIMITER)
 		p[strlen(p)-1] = '\0';
-	}
 	/* cut off leading delimiters */
-	while (p[0] == LIST_DELIMITER) {
-		count--;
+	while (p[0] == LIST_DELIMITER)
 		p++; 
-	}
 	
-	/* count delimiters to calculate elements */
-	for (count=i=0; p[i]; i++)
-		if (p[i] == LIST_DELIMITER) count++;
+	/* count delimiters to calculate the number of elements */
+	for (n = i = 0; p[i]; i++)
+		if (p[i] == LIST_DELIMITER) n++;
 	
-	if ((result = (char **)malloc(sizeof(char *) * count+2)) == NULL) {
+	if ((result = (char **)malloc(sizeof(char *) * (n + 2))) == NULL) {
 		printf ("Sorry, can't get enough memory, exiting.\n");
-		exit(0);
+		exit(1);
 	}
 	
 	/*
@@ -1285,30 +1340,28 @@
 	 * characters are different from any delimiters,
 	 * so we don't have to care about this.
 	 */
-	count = 0;
-	while ((q=strchr(p, LIST_DELIMITER))) {
-		result[count] = (char *) malloc((q - p + 1) * sizeof(char));
-		if (result[count] == NULL) {
+	m = 0;
+	while ((q = strchr(p, LIST_DELIMITER)) && m < n) {
+		result[m] = (char *) malloc((q - p + 1) * sizeof(char));
+		if (result[m] == NULL) {
 			printf ("Sorry, can't get enough memory, exiting.\n");
-			exit(0);
+			exit(1);
 		}
-		strncpy(result[count], p, q - p);
-		result[count][q - p] = '\0';
+		memcpy(result[m], p, q - p);
+		result[m][q - p] = '\0';
 		p = q; p++;
-		count++;
+		m++;
 	}
-	if ((result[count] = \
-	     (char *)malloc(sizeof(char) * strlen(p) + 1)) == NULL) {
+	if ((result[m] = strdup(p)) == NULL) {
 		printf ("Sorry, can't get enough memory, exiting.\n");
-		exit(0);
+		exit(1);
 	}
-	strcpy(result[count],p);
-	result[++count] = NULL;
+	result[++m] = NULL;
 
 #if 0
-	count=0;
-	while (result[count])
-		dprintf ("#%d: %s\n", count, StripDomains[count++]);
+	m = 0;
+	while (result[m])
+		dprintf ("#%d: %s\n", m, result[m++]);
 #endif
 	return result;
 }
@@ -1539,30 +1592,48 @@
 	int fac, prilev, lognum;
 	int msglen;
 	char *timestamp;
+#ifdef __gnu_linux__
+	sigset_t mask;
+#else
+#ifndef SYSV
+	sigset_t omask;
+#endif
+#endif
 
 	dprintf("logmsg: %s, flags %x, from %s, msg %s\n", textpri(pri), flags, from, msg);
 
+#ifdef __gnu_linux__
+	sigemptyset(&mask);
+	sigaddset(&mask, SIGHUP);
+	sigaddset(&mask, SIGALRM);
+	sigprocmask(SIG_BLOCK, &mask, NULL);
+#else
 #ifndef SYSV
 	omask = sigblock(sigmask(SIGHUP)|sigmask(SIGALRM));
 #endif
+#endif
 
 	/*
 	 * Check to see if msg looks non-standard.
+	 *
+	 * A message looks like
+	 * Nov 17 11:42:33 CRON[
+	 * 01234567890123456
+	 *    ^  ^  ^  ^  ^
+	 *
+	 * Remote messages are not accompanied by a timestamp.
+	 * Local messages are accompanied by a timestamp (program's timezone)
 	 */
 	msglen = strlen(msg);
-	if (msglen < 16 || msg[3] != ' ' || msg[6] != ' ' ||
-	    msg[9] != ':' || msg[12] != ':' || msg[15] != ' ')
-		flags |= ADDDATE;
-
-	(void) time(&now);
-	if (flags & ADDDATE)
-		timestamp = ctime(&now) + 4;
-	else {
-		timestamp = msg;
+	if (!(msglen < 16 || msg[3] != ' ' || msg[6] != ' ' ||
+	    msg[9] != ':' || msg[12] != ':' || msg[15] != ' ')) {
 		msg += 16;
 		msglen -= 16;
 	}
 
+	(void) time(&now);
+	timestamp = ctime(&now) + 4;
+
 	/* extract facility and priority level */
 	if (flags & MARK)
 		fac = LOG_NFACILITIES;
@@ -1581,9 +1652,13 @@
 			(void) close(f->f_file);
 			f->f_file = -1;
 		}
+#ifdef __gnu_linux__
+		sigprocmask(SIG_UNBLOCK, &mask, NULL);
+#else
 #ifndef SYSV
 		(void) sigsetmask(omask);
 #endif
+#endif
 		return;
 	}
 #ifdef SYSV
@@ -1646,9 +1721,13 @@
 			}
 		}
 	}
+#ifdef __gnu_linux__
+	sigprocmask(SIG_UNBLOCK, &mask, NULL);
+#else
 #ifndef SYSV
 	(void) sigsetmask(omask);
 #endif
+#endif
 }
 #if FALSE
 } /* balance parentheses for emacs */
@@ -1771,7 +1850,7 @@
 			dprintf("Not sending message to remote.\n");
 		else {
 			f->f_time = now;
-			(void) snprintf(line, sizeof(line), "<%d>%s\n", f->f_prevpri, \
+			(void) snprintf(line, sizeof(line), "<%d>%s", f->f_prevpri, \
 				(char *) iov[4].iov_base);
 			l = strlen(line);
 			if (l > MAXLINE)
@@ -1815,7 +1894,7 @@
 			v->iov_len = 1;
 		}
 	again:
-		/* f->f_file == -1 is an indicator that the we couldn't
+		/* f->f_file == -1 is an indicator that we couldn't
 		   open the file at startup. */
 		if (f->f_file == -1)
 			break;
@@ -1852,7 +1931,7 @@
 				errno = e;
 				logerror(f->f_un.f_fname);
 			}
-		} else if (f->f_flags & SYNC_FILE)
+		} else if (f->f_type == F_FILE && (f->f_flags & SYNC_FILE))
 			(void) fsync(f->f_file);
 		break;
 
@@ -1891,7 +1970,7 @@
 	register struct filed *f;
 	struct iovec *iov;
 {
-	char p[6 + UNAMESZ];
+	char p[sizeof (_PATH_DEV) + UNAMESZ];
 	register int i;
 	int ttyf, len;
 	static int reenter = 0;
@@ -1899,6 +1978,8 @@
 	struct utmp *uptr;
 	char greetings[200];
 
+	(void) &len;
+
 	if (reenter++)
 		return;
 
@@ -1913,7 +1994,6 @@
 	if (fork() == 0) {
 		(void) signal(SIGTERM, SIG_DFL);
 		(void) alarm(0);
-		(void) signal(SIGALRM, endtty);
 #ifndef SYSV
 		(void) signal(SIGTTOU, SIG_IGN);
 		(void) sigsetmask(0);
@@ -1929,7 +2009,7 @@
 			/* is this slot used? */
 			if (ut.ut_name[0] == '\0')
 				continue;
-			if (ut.ut_type == LOGIN_PROCESS)
+			if (ut.ut_type != USER_PROCESS)
 			        continue;
 			if (!(strcmp (ut.ut_name,"LOGIN"))) /* paranoia */
 			        continue;
@@ -1959,6 +2039,7 @@
 				iov[1].iov_len = 0;
 			}
 			if (setjmp(ttybuf) == 0) {
+				(void) signal(SIGALRM, endtty);
 				(void) alarm(15);
 				/* open the terminal */
 				ttyf = open(p, O_WRONLY|O_NOCTTY);
