--- sysklogd-1.4.1/Makefile~	2006-03-30 02:22:54.000000000 +0300
+++ sysklogd-1.4.1/Makefile	2006-03-30 02:24:49.000000000 +0300
@@ -3,8 +3,7 @@
 CC= gcc
 #CFLAGS= -g -DSYSV -Wall
 #LDFLAGS= -g
-CFLAGS= $(RPM_OPT_FLAGS) -O3 -DSYSV -fomit-frame-pointer -Wall -fno-strength-reduce
-# -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+CFLAGS= $(OPTIMIZE) -DSYSV -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 LDFLAGS= -s
 
 # Look where your install program is.
