--- sysklogd-1.4.1/syslogd.c	2006-03-27 22:46:40.000000000 +0300
+++ sysklogd-1.4.1.sighandler/syslogd.c	2006-03-27 22:48:30.000000000 +0300
@@ -904,11 +904,17 @@
 		dprintf("Checking pidfile.\n");
 		if (!check_pid(PidFile))
 		{
+			/*
+			 * set the handler before forking, because on very fast cpu's the
+			 * signal is delivered before the handler is installed. and then
+			 * the parent exits with non-success.
+			 */
+			signal (SIGTERM, doexit);
+
 			if (fork()) {
 				/*
 				 * Parent process
 				 */
-				signal (SIGTERM, doexit);
 				sleep(300);
 				/*
 				 * Not reached unless something major went wrong.  5
